### PR TITLE
Da/nft bridging

### PR DESCRIPTION
**Pull request by tadad**
**Problem**:
NFT bridging isn't enabled

**Solution**:
Enable it in the engine

**Notes**:
some extra fixes and logic to fungible deposits as well

### CODE REVIEW

File: lib/zig/sys/engine.hoon [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+          symbol=@t
+          properties=(pset:smart @tas)
+          supply=@ud
+          cap=(unit @ud)  ::  (~ if no cap)
+          mintable=?      ::  automatically set to %.n if supply == cap
+          minters=(pset:smart address:smart)
+          deployer=id:smart
+          salt=@
+      ==
+    --
```
Comment (tadad) [inline lib/zig/sys/engine.hoon:16]: should I import /con/compiled/nft/hoon for this mold? Or just leave it her

**Comment by dr-frmr** [comment]
Please build off this PR: https://github.com/uqbar-dao/uqbar-core/pull/454

File: lib/zig/sys/engine.hoon (hunk 1 above)
Comment (dr-frmr) [reply lib/zig/sys/engine.hoon:16]: yeah ideally import `/con/lib/nft/hoon`

File: con/lib/nft.hoon [hunk 2]
```diff
@@ -180,6 +180,8 @@
   ++  deploy
     |=  [=context act=deploy:sur]
     ^-  (quip call diff)
+    ::  make salt unique by including deployer + their input
+    =/  salt  (cat 3 salt.act id.caller.context)
```
Comment (tadad) [inline con/lib/nft.hoon:184]: if this isn't the case, then someone can reserve the salt for an L1 nft collection, and get the engine into an inconsistent state when someone tries to bridge that L1 nft collection to uqbar

**Review by dr-frmr — APPROVED** [review]

**Pull request merged**